### PR TITLE
Replace terse for loop with map and for each

### PR DESCRIPTION
**Pull request by N8python**
I replaced the for loop, which was hard to understand at a glance, with a map and a forEach, which helps get the point across. In addition, I removed the terse ++ operator, replacing it with +=, and made both object declarations multiple lines. This code, while it sacrifices a few milliseconds of performance, allows an easier understanding of what the code does, and how it uses data.

### CODE REVIEW

File: src/vs/editor/contrib/bracketMatching/bracketMatching.ts [hunk 1]
```diff
@@ -244,14 +244,18 @@ export class BracketMatchingController extends Disposable implements editorCommo
...
-			let brackets = this._lastBracketsData[i].brackets;
-			if (brackets) {
-				newDecorations[newDecorationsLen++] = { range: brackets[0], options: BracketMatchingController._DECORATION_OPTIONS };
-				newDecorations[newDecorationsLen++] = { range: brackets[1], options: BracketMatchingController._DECORATION_OPTIONS };
-			}
-		}
-
+		this._lastBracketsData.map(bracketData => bracketData.brackets).filter(brackets => brackets).forEach(brackets => {
+			newDecorations[newDecorationsLen] = {
+				range: brackets[0],
```
Comment (RDIL) [inline src/vs/editor/contrib/bracketMatching/bracketMatching.ts]: Issue - object is possibly `'null'`.  Needs fix.

File: src/vs/editor/contrib/bracketMatching/bracketMatching.ts [hunk 2]
```diff
@@ -244,14 +244,18 @@ export class BracketMatchingController extends Disposable implements editorCommo
...
+				options: BracketMatchingController._DECORATION_OPTIONS
+			};
+			newDecorationsLen += 1;
+			newDecorations[newDecorationsLen] = {
+				range: brackets[0],
```
Comment (RDIL) [inline src/vs/editor/contrib/bracketMatching/bracketMatching.ts]: Issue - object is possibly `'null'`.  Needs fix.

**Review by RDIL — CHANGES_REQUESTED** [review]
The Azure DevOps build needs to pass before maintainers will merge this.  Here is what you need to keep in mind -

**Comment by alexandrudima** [comment]
Sorry, we don't accept PRs that only tweak code style...

**Pull request closed without merging**